### PR TITLE
Spark: Check for hive support when using SparkSessionCatalog

### LINKED ISSUE #5692: Spark: Check for hive support when using SparkSessionCatalog
### Apache Iceberg version

0.14.0 (latest release)

### Query engine

Spark

### Please describe the bug 🐞

If hive support is not enabled for SparkSession, Spark will use in-memory catalog for its built-in catalog. This leads to Iceberg's SparkSessionCatalog delegating requests to the in-memory catalog, which can be confusing since user expects requests to be handled by hive catalog. 
For example, consider a hive metastore at thrift://localhost:9093 with namespaces:
```
+---------+
|namespace|
+---------+
|       db|
|  default|
+---------+
```
and a table in namespace "db":
```
+---------+---------+-----------+
|namespace|tableName|isTemporary|
+---------+---------+-----------+
|       db|    table|      false|
+---------+---------+-----------+
```
The following steps will produce inconsistent results:
1. Build a SparkSession without hive support:
```
SparkSession spark = SparkSession
        .builder()
        .appName("SparkHiveTest")
        .master("local[*]")
        .config("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
        .config("spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog")
        .config("spark.sql.catalog.spark_catalog.type", "hive")
        .config("spark.sql.catalog.spark_catalog.uri", "thrift://localhost:9083")
        .config("spark.sql.catalog.spark_catalog.warehouse", "file:///tmp/spark-warehouse")
        .getOrCreate();
```
2. List namespaces, which will be handled by spark's in-memory catalog, therefore not showing namespace "db":
```
spark.sql("SHOW DATABASES").show();

+---------+
|namespace|
+---------+
|  default|
+---------+
```
3. Create namespace "db", which will be handled by spark's in-memory catalog, therefore succeeds:
```
spark.sql("CREATE DATABASE db");
```
4. Create table "db.table", which will be handled by iceberg's hive catalog, where "db.table" already exists, therefore fails:
```
spark.sql("CREATE TABLE db.table (id bigint, data string) USING ICEBERG"); // fails: Table db.table already exists
```
However, user expects step 4 to succeed, since "db" is a new empty namespace from user's perspective.

To avoid such problems, we should add a check to ensure hive support for spark is enabled when initializing SparkSessionCatalog.

### PR DESCRIPTION
**Pull request by linyanghao**
This resolves #5692

### CODE REVIEW

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java [hunk 1]
```diff
@@ -289,6 +292,19 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
     this.createOrcAsIceberg = options.getBoolean("orc-enabled", createOrcAsIceberg);
   }
 
+  private void checkHiveSupport() {
+    RuntimeConfig runtimeConfig = SparkSession.active().conf();
+    String sparkCatalogImpl = runtimeConfig.get(StaticSQLConf.CATALOG_IMPLEMENTATION().key());
+    if (sparkCatalogImpl != null) {
+      Preconditions.checkArgument(
+          sparkCatalogImpl.equals("hive"),
+          "Please enable hive support for Spark. "
```
Comment (nastra) [inline spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java:301]: should the error msg mention how to enable it? `Please enable hive support for Spark by setting 'spark.sql.catalogImplementation=hive'` or something along those lines

**Review by nastra — COMMENTED** [review]
changes make sense to me, but I'm unsure about their implications. @RussellSpitzer could you review this one as well please?

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java (hunk 1 above)
Comment (linyanghao) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java:301]: > should the error msg mention how to enable it? `Please enable hive support for Spark by setting 'spark.sql.catalogImplementation=hive'` or something along those lines

How about `Please enable hive support for Spark by calling enableHiveSupport() or setting 'spark.sql.catalogImplementation=hive'`

Comment (nastra) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java:301]: sgtm

**Pull request closed without merging**